### PR TITLE
Optimize string comparison with typeof operation

### PR DESCRIPTION
**Pull request by clover2123**
Signed-off-by: HyukWoo Park <hyukwoo.park@samsung.com>

### CODE REVIEW

**Comment by clover2123** [comment]
I measured the performance of this patch compared to origin for the test code below

```js
(function() {
    var x = undefined;
    var res;
    for (var i = 0; i < 100000000; i++) {
        res = ("undefined" === typeof x);
    }
    return res;
})();
```

Origin: 2700ms
Patch: 1175ms

**Comment by ksh8281** [comment]
Could you test octane & sunspider performance and loadtime?

**Comment by clover2123** [comment]
@ksh8281 
Octane score decreased. I think that some more updates are required for this patch.

**Comment by clover2123** [comment]
I just tuned the performane and got the following result
2251 -> 2265 (octane)

File: src/runtime/Value.cpp [hunk 1]
```diff
@@ -301,13 +301,15 @@ bool Value::abstractEqualsToSlowCase(ExecutionState& state, const Value& val) co
...
-        if (std::isnan(a) || std::isnan(b))
+        if (UNLIKELY(std::isnan(a) || std::isnan(b)))
             return false;
-        else if (a == b)
-            return true;
 
-        return false;
+        return a == b;
     } else {
+        if (u.asInt64 == val.u.asInt64) {
```
Comment (bbrto21) [inline src/runtime/Value.cpp:309]: Q. Can you explain why this is possible?

Comment (clover2123) [reply src/runtime/Value.cpp:309]: If two operands have the same value, it means that they are equal.
So, return true directly without further checks below.

**Review by bbrto21 — APPROVED** [review]

**Review by ksh8281 — APPROVED** [review]

**Pull request merged**